### ISSUE TITLE
mustPayToPubKeyAddress and mustPayWithDatumToPubKeyAddress have misleading docstring

### ISSUE THREAD

**Issue by nazrhom** [Plutonomicon/cardano-transaction-lib#982]
The two functions above use the following docstring:

```
-- | Lock the value with a payment public key hash and (optionally) a stake
-- | public key hash.
```
however the argument they take is a `StakePubKeyHash` rather than a `Maybe StakePubKeyHash`. 

I noticed older version of ledger-constraints used to have the same wording, but they have recently updated it (crf. https://github.com/input-output-hk/plutus-apps/blob/main/plutus-ledger-constraints/src/Ledger/Constraints/TxConstraints.hs#L310-L312)

**Comment by zmrocze** [comment]
Doesn't apply already:
 - ["public key address"](https://github.com/Plutonomicon/cardano-transaction-lib/blob/09174df2de288cb27e001cc437281d362024db90/src/Internal/Types/TxConstraints.purs#L284) for address with stake keys (base)
 - ["public key"](https://github.com/Plutonomicon/cardano-transaction-lib/blob/09174df2de288cb27e001cc437281d362024db90/src/Internal/Types/TxConstraints.purs#L332) or "public key hash" for address without stake (enterprise)
 
 Officially these are called base and enterprise addresses, but noone uses these names.

**Comment by Bradley-Heather** [comment]
Is there any reason this can't be closed? 

Perhaps we could add the below just for clarity.
```
-- | Lock the value with a public key address. (Base Address)
-- | Lock the value with a public key. (Enterprise Address)
```

**Comment by Bradley-Heather** [comment]
closed by #1303
